src/hotspot/share/gc/shenandoah/shenandoahRootProcessor.hpp

```diff
@@ -107,23 +107,21 @@ class ShenandoahCodeCacheRoots {
   void code_blobs_do(CodeBlobClosure* blob_cl, uint worker_id);
 };
 
-template <bool CONCURRENT, bool SINGLE_THREADED>
+template <bool CONCURRENT>
 class ShenandoahClassLoaderDataRoots {
 private:
   ShenandoahSharedSemaphore     _semaphore;
   ShenandoahPhaseTimings::Phase _phase;
 
   static uint worker_count(uint n_workers) {
-    if (SINGLE_THREADED) return 1u;
-
     // Limit concurrency a bit, otherwise it wastes resources when workers are tripping
     // over each other. This also leaves free workers to process other parts of the root
     // set, while admitted workers are busy with doing the CLDG walk.
     return MAX2(1u, MIN2(ShenandoahSharedSemaphore::max_tokens(), n_workers / 2));
   }
 
 public:
-  ShenandoahClassLoaderDataRoots(ShenandoahPhaseTimings::Phase phase, uint n_workers);
+  ShenandoahClassLoaderDataRoots(ShenandoahPhaseTimings::Phase phase, uint n_workers, bool heapdump);
```

Comment:
Can we call the new parameter `heap_iteration`? This would match the `ShenandoahHeapIterationRootScanner` name. I could also imagine that iteration is used for something else than a heap dump (maybe JFR/JVMTI heap walk for profiling/debugging?).

Reply:
Fixed. Thanks!